modules/dag-l1/src/main/scala/io/constellationnetwork/dag/l1/infrastructure/address/storage/AddressStorage.scala

```diff
@@ -12,23 +12,37 @@ import eu.timepit.refined.auto._
 
 object AddressStorage {
   def make[F[_]: Async]: F[AddressStorage[F]] =
-    Ref.of[F, Map[Address, Balance]](Map.empty).map(make(_))
+    (Ref.of[F, Map[Address, Balance]](Map.empty), Ref.of[F, Map[Address, Balance]](Map.empty)).mapN((balances, tokenLockBalances) =>
+      make(balances, tokenLockBalances)
+    )
 
-  def make[F[_]: Async](balances: Map[Address, Balance]): F[AddressStorage[F]] =
-    Ref.of[F, Map[Address, Balance]](balances).map(make(_))
+  def make[F[_]: Async](balances: Map[Address, Balance], tokenLockBalances: Map[Address, Balance]): F[AddressStorage[F]] =
+    (Ref.of[F, Map[Address, Balance]](balances), Ref.of[F, Map[Address, Balance]](tokenLockBalances)).mapN((balances, tokenLockBalances) =>
+      make(balances, tokenLockBalances)
+    )
 
-  def make[F[_]: Async](balances: Ref[F, Map[Address, Balance]]): AddressStorage[F] =
+  def make[F[_]: Async](balances: Ref[F, Map[Address, Balance]], tokenLockBalances: Ref[F, Map[Address, Balance]]): AddressStorage[F] =
     new AddressStorage[F] {
-      def getState: F[Map[Address, Balance]] =
-        balances.get
+      def getState: F[(Map[Address, Balance], Map[Address, Balance])] = for {
+        walletBalances <- balances.get
+        walletTokenLockBalances <- tokenLockBalances.get
+      } yield (walletBalances, walletTokenLockBalances)
 
       def getBalance(address: Address): F[Balance] =
         balances.get.map(_.getOrElse(address, Balance.empty))
 
+      def getTokenLockBalance(address: Address): F[Balance] =
+        tokenLockBalances.get.map(_.getOrElse(address, Balance.empty))
+
       def updateBalances(addressBalances: Map[Address, Balance]): F[Unit] =
         balances.update(_ ++ addressBalances)
 
-      def clean: F[Unit] =
+      def updateTokenLockBalances(tokenLockAddressBalances: Map[Address, Balance]): F[Unit] =
+        tokenLockBalances.update(_ ++ tokenLockAddressBalances)
+
+      def clean: F[Unit] = {
         balances.set(Map.empty)
+        tokenLockBalances.set(Map.empty)
```

Comment:
The `balances.set` won't be called. It is like calling:

```
Promise(1)
Promise(2)
```
instead of
```
Promise(1).then(a => Promise(2))
```

So if you have 2 `F[_]` effects then you should chain them:
```scala
balances.set(Map.empty) >> tokenLockBalances.set(Map.empty)
```

Reply:
👍🏼